### PR TITLE
Add human-readable weaver-cli output with auto-resolve (Phase 1)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Implement Phase 1 of the human-readable weaver-cli output subsystem with an auto-resolving --output flag (auto, human, json) based on stdout being a TTY.
- Refactor unit tests to remove duplication and align with the new test structure.
- Extend test coverage and end-to-end tests to exercise the new rendering path and auto-resolve behavior.
- Upgrade test tooling (rstest-bdd family) to align with the new test structure.

📎 **Task**: https://www.devboxer.com/task/fd1c9cd0-f6c8-472c-9e31-5b2b8e4d840d

### Code
- Add a new output subsystem under crates/weaver-cli/src/output with modules: mod.rs, models.rs, render.rs, source.rs
- Introduce OutputFormat, ResolvedOutputFormat, and OutputContext to model and drive rendering
- Extend the CLI to accept a new --output flag and resolve Auto based on stdout
- Wire read_daemon_messages to consume OutputSettings (format + context) and render human output when requested
- Update IoStreams to track whether stdout is a terminal and expose that information for resolution
- Implement render_human_output with domain/operation dispatching for:
  - observe get-definition
  - observe find-references
  - verify diagnostics
  - act (verification failures)
- Add source-resolution helpers (from_uri, from_path_or_uri) and URI argument extraction for fallback rendering
- Wire integration points in lib.rs/main.rs to pass along OutputFormat/OutputContext to the daemon path

### Tests
- Refactor tests to remove duplication and align with the new test structure; extend unit/behaviour tests to exercise new rendering path and auto-resolve behavior
- Add test support for creating temporary sources and test fixtures for daemon payloads
- Add new test suites and features for human-readable output rendering:
  - crates/weaver-cli/tests/features/weaver_cli_output.feature
  - Extended behaviour tests in crates/weaver-cli/src/tests/behaviour.rs
  - End-to-end tests in crates/weaver-e2e/tests/human_output.rs

### Documentation
- Add docs/execplans/phase-1-human-readable-output.md describing Phase 1 plan, acceptance criteria, and design decisions
- Update roadmap and user guides to reflect the new output mode and human rendering expectations
- Update docs/weaver-design.md to document the new --output flag and rendering approach

### End-to-end
- Extend weaver-e2e to validate human-readable rendering for definitions and diagnostics
- Include scenarios that exercise missing source content fallbacks and raw JSON output mode

### Dependencies
- Upgrade rstest-bdd and related macros across the workspace to v0.4.0 to align with new test structure

### Notes
- This PR implements Phase 1 of human-readable output rendering. Further improvements and edge-case handling will be addressed in subsequent PRs.

📎 **Task**: https://www.devboxer.com/task/fd1c9cd0-f6c8-472c-9e31-5b2b8e4d840d

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/output/mod.rs](https://github.com/leynos/weaver/pull/56/files/7ff0c500ac95715ab369c41a9e3edd446c688901#diff-6e88de12525ed50e12267587d5fe3064d976cbd46c3e91ac19fad7edfd19265b)

Comment on lines +109 to +126

~~~rust
fn render_definitions(payload: &str) -> Option<String> {
    let definitions = parse_definitions(payload)?;
    if definitions.is_empty() {
        return Some(String::from("no definitions found\n"));
    }
    let locations: Vec<SourceLocation> = definitions
        .into_iter()
        .map(|definition| {
            from_uri(
                &definition.uri,
                Some(definition.line),
                Some(definition.column),
                "definition",
            )
        })
        .collect();
    Some(render::render_locations(&locations))
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Foutput%2Fmod.rs&method=)
The module contains 2 functions with similar structure: render_definitions,render_references

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/output/render.rs](https://github.com/leynos/weaver/pull/56/files/7ff0c500ac95715ab369c41a9e3edd446c688901#diff-a7379616e386906d3d029e58e610fb3dacc87ea4230a16f96cb6117fc1780ebf)

Comment on lines +13 to +63

~~~rust
pub(crate) fn render_locations(locations: &[SourceLocation]) -> String {
    if locations.is_empty() {
        return String::new();
    }

    let mut output = String::new();
    let mut order: Vec<String> = Vec::new();
    let mut grouped: HashMap<String, Vec<&SourceLocation>> = HashMap::new();

    for location in locations {
        let key = location.source.display();
        if !grouped.contains_key(&key) {
            order.push(key.clone());
        }
        grouped.entry(key).or_default().push(location);
    }

    for (group_index, key) in order.iter().enumerate() {
        if group_index > 0 {
            output.push('\n');
        }
        let Some(group) = grouped.get(key) else {
            continue;
        };
        let source = &group[0].source;
        writeln!(output, "{key}").expect("write header");

        let content_result = source
            .as_path()
            .map(|path| fs::read_to_string(path).map_err(|err| err.to_string()));

        for (index, location) in group.iter().enumerate() {
            if index > 0 {
                output.push('\n');
            }
            match content_result.as_ref() {
                Some(Ok(content)) => render_location_block(&mut output, location, Some(content)),
                Some(Err(error)) => {
                    render_unresolved(
                        &mut output,
                        location,
                        format!("source unavailable: {error}"),
                    );
                }
                None => render_location_block(&mut output, location, None),
            }
        }
    }

    output
}
~~~

❌ New issue: [Complex Method](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Complex+Method&filename=crates%2Fweaver-cli%2Fsrc%2Foutput%2Frender.rs&method=render_locations)
render_locations has a cyclomatic complexity of 9, threshold = 9

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/output/render.rs](https://github.com/leynos/weaver/pull/56/files/7ff0c500ac95715ab369c41a9e3edd446c688901#diff-a7379616e386906d3d029e58e610fb3dacc87ea4230a16f96cb6117fc1780ebf)

Comment on lines +13 to +63

~~~rust
pub(crate) fn render_locations(locations: &[SourceLocation]) -> String {
    if locations.is_empty() {
        return String::new();
    }

    let mut output = String::new();
    let mut order: Vec<String> = Vec::new();
    let mut grouped: HashMap<String, Vec<&SourceLocation>> = HashMap::new();

    for location in locations {
        let key = location.source.display();
        if !grouped.contains_key(&key) {
            order.push(key.clone());
        }
        grouped.entry(key).or_default().push(location);
    }

    for (group_index, key) in order.iter().enumerate() {
        if group_index > 0 {
            output.push('\n');
        }
        let Some(group) = grouped.get(key) else {
            continue;
        };
        let source = &group[0].source;
        writeln!(output, "{key}").expect("write header");

        let content_result = source
            .as_path()
            .map(|path| fs::read_to_string(path).map_err(|err| err.to_string()));

        for (index, location) in group.iter().enumerate() {
            if index > 0 {
                output.push('\n');
            }
            match content_result.as_ref() {
                Some(Ok(content)) => render_location_block(&mut output, location, Some(content)),
                Some(Err(error)) => {
                    render_unresolved(
                        &mut output,
                        location,
                        format!("source unavailable: {error}"),
                    );
                }
                None => render_location_block(&mut output, location, None),
            }
        }
    }

    output
}
~~~

❌ New issue: [Bumpy Road Ahead](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Bumpy+Road+Ahead&filename=crates%2Fweaver-cli%2Fsrc%2Foutput%2Frender.rs&method=render_locations)
render_locations has 3 blocks with nested conditional logic. Any nesting of 2 or deeper is considered. Threshold is 2 blocks per function

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/tests/behaviour.rs](https://github.com/leynos/weaver/pull/56/files/7ff0c500ac95715ab369c41a9e3edd446c688901#diff-0c0c723392282dde77b205be25ccf15bd77b49a3f09a2a7c8b11940c9ca4bb41)

Comment on lines +145 to +156

~~~rust
fn when_operator_runs_definition(world: &RefCell<TestWorld>) {
    let uri = world
        .borrow()
        .source_uri()
        .expect("source uri missing")
        .to_owned();
    let command = format!("--output human observe get-definition --uri {uri} --position 2:5");
    world
        .borrow_mut()
        .run(&command)
        .expect("failed to run definition command");
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Fbehaviour.rs&method=)
The module contains 6 functions with similar structure: then_stderr_contains,then_stdout_contains,then_stdout_does_not_contain,when_operator_runs_definition and 2 more functions

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/tests/behaviour.rs](https://github.com/leynos/weaver/pull/56/files/7ff0c500ac95715ab369c41a9e3edd446c688901#diff-0c0c723392282dde77b205be25ccf15bd77b49a3f09a2a7c8b11940c9ca4bb41)

Comment on lines +145 to +156

~~~rust
fn when_operator_runs_definition(world: &RefCell<TestWorld>) {
    let uri = world
        .borrow()
        .source_uri()
        .expect("source uri missing")
        .to_owned();
    let command = format!("--output human observe get-definition --uri {uri} --position 2:5");
    world
        .borrow_mut()
        .run(&command)
        .expect("failed to run definition command");
}
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Fbehaviour.rs&method=)
The module contains 6 functions with similar structure: then_stderr_contains,then_stdout_contains,then_stdout_does_not_contain,when_operator_runs_definition and 2 more functions

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/tests/unit.rs](https://github.com/leynos/weaver/pull/56/files/7ff0c500ac95715ab369c41a9e3edd446c688901#diff-278854716102c6463c98bae84f8e597c2fc78c2a13f4df2de9c4e9cdbcf159ac)

Comment on lines +120 to +130

~~~rust
    let mut io = IoStreams::with_terminal_status(&mut stdout, &mut stderr, false);
    let context = OutputContext::new("observe", "get-definition", Vec::new());
    let error = read_daemon_messages(
        &mut cursor,
        &mut io,
        OutputSettings {
            format: ResolvedOutputFormat::Json,
            context: &context,
        },
    )
    .unwrap_err();
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Ftests%2Funit.rs&method=)
The module contains 2 functions with similar structure: read_daemon_messages_errors_without_exit,read_daemon_messages_warns_after_empty_lines

File: crates/weaver-cli/src/output/render.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                },
+            );
+        }
+    }
+}
+
+fn render_caret_line(output: &mut String, context: CaretContext<'_>) {
+    let line_len = context.text.chars().count();
+    let column_index = context.column.saturating_sub(1) as usize;
+    let caret_pos = column_index.min(line_len);
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/weaver-cli/src/output/render.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[crates/weaver-cli/src/lib.rs](https://github.com/leynos/weaver/pull/56/files/d1f0ce21c95509a383a9b2062f830ec0689091b3#diff-17a30b0584d2348541b6621fe97320b4f869125a087efff03bf2775b20245ba9)

Comment on lines +65 to +71

~~~rust
    pub(crate) fn new(stdout: &'a mut W, stderr: &'a mut E, stdout_is_terminal: bool) -> Self {
        Self {
            stdout,
            stderr,
            stdout_is_terminal,
        }
    }
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/71837/delta?repo-id=1205553&review-id=56&biomarker=Code+Duplication&filename=crates%2Fweaver-cli%2Fsrc%2Flib.rs&method=)
The module contains 2 functions with similar structure: IoStreams.new,IoStreams.with_terminal_status

File: crates/weaver-cli/src/lib.rs [hunk 2]
```diff
@@ -53,11 +58,33 @@ const EMPTY_LINE_LIMIT: usize = 10;
...
+        Self {
+            stdout,
+            stderr,
+            stdout_is_terminal,
+        }
+    }
+
+    pub(crate) const fn stdout_is_terminal(&self) -> bool {
+        self.stdout_is_terminal
     }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/weaver-cli/src/lib.rs:75]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: crates/weaver-cli/src/runtime_utils.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    stdout.flush().map_err(AppError::EmitCapabilities)
+}
+
+pub(crate) fn exit_code_from_status(status: i32) -> ExitCode {
+    if (0..=255).contains(&status) {
+        ExitCode::from(status as u8)
+    } else {
+        ExitCode::FAILURE
+    }
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/weaver-cli/src/runtime_utils.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: crates/weaver-cli/Cargo.toml [hunk 4]
```diff
@@ -15,6 +15,8 @@ serde_json = { workspace = true }
 ortho_config = { workspace = true }
 thiserror = { workspace = true }
 cap-std = { workspace = true }
+url = { workspace = true }
+unicode-width = "0.2.2"
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/weaver-cli/Cargo.toml]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.
